protocols/autonat/tests/test_client.rs

```diff
@@ -497,7 +489,8 @@ async fn test_outbound_failure() {
         }
 
         let inactive = servers.split_off(1);
-        // Drop the handles of the inactive servers to kill them.
+
+        #[allow(clippy::needless_collect)] // Drop the handles of the inactive servers to kill them.
```

Comment:
out of curiosity to understand in this case why clippy was complaining I commented the `allow` and ran  the `cargo +nightly clippy --all-features --all-targets -- -A clippy::type_complexity -A clippy::pedantic -D warnings` command. But clippy didn't complain, is it not required?

Reply:
You need the `--workspace` flag! My best guess is that this is because our root Cargo.toml is a crate _and_ a workspace declaration.

I wonder whether we could avoid this if we put the meta crate in a directory too.

Thoughts @mxinden?

Reply:
yeah, was running outdated rust version :sweat_smile: sorry. Btw with the latest nightly there are new lints triggered for [`large_enum_variant`](https://rust-lang.github.io/rust-clippy/master/index.html#large_enum_variant).

Reply:
> yeah, was running outdated rust version 😅 sorry. Btw with the latest nightly there are new lints triggered for [`large_enum_variant`](https://rust-lang.github.io/rust-clippy/master/index.html#large_enum_variant).

Yeah I know :)

I opened https://github.com/libp2p/rust-libp2p/issues/2951 for this recently!